### ISSUE TITLE
Memory leak addon networks

### ISSUE THREAD

**Issue by brasilcraft54** [Sefiraat/Networks#65]
my server is running out of memory after I add addon networks.

information: https://prnt.sc/HWnWdyePrXh6 ( more Networks - DEV 34 (git eb71c) ) I removed it so it is not in this image.

my memory: https://prnt.sc/FmuCsX4cCYQp

I managed to get a report by spark a few minutes before it crashed: https://spark.lucko.me/JxveHzB2IV

I spent 2 days with the server running out of memory and I had to restart, then I did a test on 04/10/2022 and left the server without the addon networks for 24 hours and I didn't get the problem anymore.
and that is how I confirmed that the networks were leaking memory, my server has 30 - 40 players.

**Comment by Sefiraat** [comment]
In order to diagnose what in Networks is causing you an issue please get a heapdump while the issue is occurring. This should ideally be done on a test server so as to not cause an issue on your live environment. When this is done, please paste the result here. Thanks.

**Comment by brasilcraft54** [comment]
> In order to diagnose what in Networks is causing you an issue please get a heapdump while the issue is occurring. This should ideally be done on a test server so as to not cause an issue on your live environment. When this is done, please paste the result here. Thanks.

I have already seen above in spark that was minutes before he ran out of memory.

**Comment by Sefiraat** [comment]
> > In order to diagnose what in Networks is causing you an issue please get a heapdump while the issue is occurring. This should ideally be done on a test server so as to not cause an issue on your live environment. When this is done, please paste the result here. Thanks.
> 
> I have already seen above in spark that was minutes before he ran out of memory.

that doesn’t address anything I said?

**Comment by brasilcraft54** [comment]
> Also please ensure that players build is disabled before the next test as stated before. This will be the 6th time of saying this and it needs to be done before I’ll look at this further as I’m struggling to triage here.

there are about 10 people with the networks ready

**Comment by brasilcraft54** [comment]
> Also please ensure that players build is disabled before the next test as stated before. This will be the 6th time of saying this and it needs to be done before I’ll look at this further as I’m struggling to triage here.

another thing i don't understand what you want you said 6 times but i don't understand you want me to remove the networks of all players?

**Comment by Sefiraat** [comment]
> > Also please ensure that players build is disabled before the next test as stated before. This will be the 6th time of saying this and it needs to be done before I’ll look at this further as I’m struggling to triage here.
> 
> there are about 10 people with the networks ready
 
Please, we went to a players base and I said how their build was very bad and it needs fixing. Please actually just do the things I am asking and stop looking for loopholes or shortcuts to the triaging process? I want to help you but you are almost actively trying to stop me moving forward at this point.

**Comment by Sefiraat** [comment]
To reiterate, 

- Have that network we say disabled or the cells removed 
- Set up your test server to match live (any large server like yours should have a valid test setup)
- In your test server, while the issue is occurring, run a heapdump and pop the results in here.

**Comment by brasilcraft54** [comment]
> 

I'm not looking for shortcuts but I can't understand, and also it's not only that player that owns it now how will I know all the players that own that network?

**Comment by Sefiraat** [comment]
> I'm not looking for shortcuts but I can't understand, and also it's not only that player that owns it now how will I know all the players that own that network?

No one said all networks? Just that one that had a visible issue. The cell limit will be added in a future version but at least we can rule out the cells being an issue if you sort it.

**Comment by brasilcraft54** [comment]
> 

if I don't have a limit another player will do it without me knowing and the problem can happen again.

**Comment by Sefiraat** [comment]
As you have not completed any of the steps I've asked for and on checking no other server I work with is reporting this issue, there is little I can do on this and will need to close it. When you are happy to follow the steps or if you find out more information, please open a new issue and I'll happily take a look.
